rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -510,12 +511,17 @@ private ABICallVoteResult executeVoteFederationChangeFunction(boolean dryRun, AB
      */
     private Integer createPendingFederation(boolean dryRun) {
         PendingFederation currentPendingFederation = provider.getPendingFederation();
-
         if (currentPendingFederation != null) {
             logger.warn("[createPendingFederation] A pending federation already exists.");
             return FederationChangeResponseCode.PENDING_FEDERATION_ALREADY_EXISTS.getCode();
         }
 
+        Optional<Federation> currentProposedFederation = provider.getProposedFederation(constants, activations);
+        if (currentProposedFederation.isPresent()) {
```

Comment:
Not a big deal, buy maybe put this in a method? `boolean proposedFederationExists`. Maybe it's useful to reuse in other places

Reply:
Done